src/librustc_mir/util/liveness.rs

```diff
@@ -47,18 +47,18 @@ use rustc::ty::TyCtxt;
 use std::io::{self, Write};
 use transform::MirSource;
 
-pub type LocalSet = IdxSetBuf<Local>;
+pub type LocalSet<V: Idx> = IdxSetBuf<V>;
 
 /// This gives the result of the liveness analysis at the boundary of
 /// basic blocks. You can use `simulate_block` to obtain the
 /// intra-block results.
-pub struct LivenessResult {
+pub struct LivenessResult<V: Idx> {
```

Comment:
We should probably add some kind of trait like `LiveVariableMap`, which would look kind of like so:

```rust
trait LiveVariableMap {
    type LiveVar;

    fn from_local(&self, local: Local) -> Option<Self::LiveVar>;
    fn from_live_var(&self, local: Self::LiveVar) -> Local;
}
```

We would also make an "identity map" type that we use for now:

```rust
struct IdentityMap;

impl LiveVariableMap for IdentityMap {
    type LiveVar = Local;

    fn from_local(&self, local: Local) -> Option<Local> {
        Some(local)
    }

    fn from_live_var(&self, local: Self::LiveVar) -> Local {
        local
    }
}
```